flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1012,10 +1016,34 @@ public void disconnectResourceManager(Exception cause) {
 		}
 	}
 
+	@Override
+	public CompletableFuture<ThreadDumpInfo> requestThreadDump(Time timeout) {
+		final Collection<ThreadInfo> threadDump = JvmUtils.createThreadDump();
+
+		final Collection<ThreadDumpInfo.ThreadInfo> threadInfos = threadDump.stream()
+			.map(threadInfo -> ThreadDumpInfo.ThreadInfo.create(threadInfo.getThreadName(), threadInfo.toString()))
+			.collect(Collectors.toList());
+
+		return CompletableFuture.completedFuture(ThreadDumpInfo.create(threadInfos));
+	}
+
 	// ======================================================================
 	//  Internal methods
 	// ======================================================================
 
+	private CompletableFuture<TransientBlobKey> putTransientBlobStream(InputStream inputStream, String fileTag) {
+		final TransientBlobCache transientBlobService = blobCacheService.getTransientBlobService();
+		final TransientBlobKey transientBlobKey;
+
+		try {
+			transientBlobKey = transientBlobService.putTransient(inputStream);
+		} catch (IOException e) {
+			log.debug("Could not upload file {}.", fileTag, e);
+			return FutureUtils.completedExceptionally(new FlinkException("Could not upload file " + fileTag + '.', e));
+		}
+		return CompletableFuture.completedFuture(transientBlobKey);
+	}
+
```

Comment:
hi @tillrohrmann, since doesn't make thread dump as stream in your design right now, we can revert it.

Reply:
I think this change is still good since it breaks up a longer method into smaller units.

Reply:
They still belong into a separate commit though.

Reply:
> I think this change is still good since it breaks up a longer method into smaller units.

ok

Reply:
Technically yes but I think it is ok to not twiddle with this commit because the benefit would be marginal.